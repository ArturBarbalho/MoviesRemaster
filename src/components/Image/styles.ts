import styled,{css} from 'styled-components'
import media from 'styled-media-query'


const Modifiers = {
    small:()=>css`
    width:350px;
    @media(max-width:450px){
        width:205px;
    } 
    @media(max-width:400px){
        width:160px;
    } 
    `,
    medium:()=>css`
    width:50%;
    max-width:500px;
    `,
    large:()=>css`
    width:100%;
    height:100vh;
    object-fit:cover;
    `,
    larger:()=>css`
    width:100vw;
    height: 100vh;
    `,
}

type props = {
    size:'small'|'medium'|'large'|'larger'
}

export const Wrapper = styled.img<props>`
${({size})=>css`
${!!size && Modifiers[size]()}
`}
`