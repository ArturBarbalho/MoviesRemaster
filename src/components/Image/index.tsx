import * as S from './styles'
type props = {
  src:string
  size:'small'|'medium'|'large'|'larger'
}
const Image = ({src, size}:props) => (
  <S.Wrapper size={size} src={src} />
    
 
)

export default Image