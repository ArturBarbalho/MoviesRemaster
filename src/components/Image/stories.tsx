import { Story, Meta } from '@storybook/react/types-6-0'
import Image from '.'

export default {
  title: 'Image',
  component: Image,
} as Meta

export const Default: Story = () => <Image  
size='larger'
src='https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80' />