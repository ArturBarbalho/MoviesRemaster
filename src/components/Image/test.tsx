import { render, screen } from '@testing-library/react'

import Image from '.'

describe('<Image />', () => {
  const src = 'https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80'

  it('should render the small size', () => {
    const { container } = render(<Image
      size='small'
      src={src}
      />)

    expect(screen.getByRole('img')).toHaveStyle(
      'width:350px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })


  it('should render the medium size', () => {
    const { container } = render(<Image
      size='medium'
      src={src}
      />)

    expect(screen.getByRole('img')).toHaveStyle(
      'max-width:500px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })


  it('should render the large size', () => {
    const { container } = render(<Image
      size='large'
      src={src}
      />)

    expect(screen.getByRole('img')).toHaveStyle(
      'width:100%;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })


  it('should render the larger size', () => {
    const { container } = render(<Image
      size='larger'
      src={src}
      />)

    expect(screen.getByRole('img')).toHaveStyle(
      'width:100vw;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})