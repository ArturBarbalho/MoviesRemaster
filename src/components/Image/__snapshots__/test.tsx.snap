// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render the large size 1`] = `
<img
  class="styles__Wrapper-sc-y6e3xk-0 kFeMay"
  src="https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80"
/>
`;

exports[`<Image /> should render the larger size 1`] = `
<img
  class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
  src="https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80"
/>
`;

exports[`<Image /> should render the medium size 1`] = `
<img
  class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
  src="https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80"
/>
`;

exports[`<Image /> should render the small size 1`] = `
<img
  class="styles__Wrapper-sc-y6e3xk-0 fOWEhk"
  src="https://images.unsplash.com/photo-1671160033378-4e6329f56555?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80"
/>
`;
